Samples/PushNotifications/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,205 @@
+            {
+                LOG_HR_MSG(
+                    args.extendedError,
+                    "The channel request is in back-off retry mode because of a retryable error! Expect delays in acquiring it. RetryCount = %d",
+                    args.retryCount);
+            }
+        });
+
+    auto result = co_await channelOperation;
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        auto channel = result.Channel();
+
+        DateTimeFormatter formater = DateTimeFormatter(L"on {month.abbreviated} {day.integer(1)}, {year.full} at {hour.integer(1)}:{minute.integer(2)}:{second.integer(2)}");
+
+        std::cout << "Channel Uri: " << winrt::to_string(channel.Uri().ToString()) << std::endl;
+        std::wcout << L"Channel Uri will expire " << formater.Format(channel.ExpirationTime()).c_str() << std::endl;
+
+        // Caller's responsibility to keep the channel alive
+        co_return channel;
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        std::cout << result.ExtendedError() << "We hit a critical non-retryable error with channel request!" << std::endl;
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task = RequestChannelAsync();
+    if (task.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result = task.GetResults();
+    return result;
+}
+
+// Register Push Event for Foreground
+void RegisterForegroundNotificationsHandler(const winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel& channel)
+{
+    winrt::event_token token = channel.PushReceived([](auto const&, PushNotificationReceivedEventArgs const& args)
+        {
+            auto payload = args.Payload();
+
+            // Do stuff to process the raw payload
+            std::string payloadString(payload.begin(), payload.end());
+            std::cout << "Push notification content received from FOREGROUND: " << payloadString << std::endl;
+
+            // Set handled to true to prevent background activation
+            args.Handled(true);
+        });
+}
+
+int main()
+{
+    // An unpackaged app must initialize dynamic dependencies so we can consume the Windows App SDK APIs.
+    HRESULT loadWindowsAppSDKHr = LoadWindowsAppSDK();
+    if (FAILED(loadWindowsAppSDKHr))
+    {
+        std::wcout << "Could not load Windows App SDK!" << std::endl;
+        return 1;
+    }
+
+    PushNotificationActivationInfo info(PushNotificationRegistrationActivators::ProtocolActivator);
+    PushNotificationManager::RegisterActivator(info);
+    // We do not call PushNotificationManager::UnregisterActivator
```

Comment:
Probably we can say that "Call Unregister once you don't need push notifications. Otherwise, we wouldn't ..."